### PR TITLE
issue #75- [F8]: Add connection checker to external recommender settings.

### PR DESCRIPTION
**Pull request by katikireddy622**
**What's in the PR**

Instead of implementing a separate button for checking the external connection we have used the existing save button to verify the form validation. We have developed a custom form validator to remote URL which takes URL as an input and establishes a connection and extracts the headers from the opened connection. From the headers, we receive a status code which when equals to HTTP_OK(200), the connection is successful. If the connection is not successfully established, the form won't be saved and returns an error message with the appropriate status code.

The implementation can be seen from the screenshots below: 

Screenshot (29)

Screenshot (30)

### CODE REVIEW

**Comment by katikireddy622** [comment]
Please Review our pull request @munterkalmsteiner

File: inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImplQualifierIntegrationTest.java [hunk 1]
```diff
@@ -82,7 +87,6 @@
     private KBConcept concept;
     private KBProperty property;
     private KBHandle conceptHandle;
-    private KBHandle propertyHandle;
```
Comment (munterkalmsteiner) [inline inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImplQualifierIntegrationTest.java:85]: Deleting this leads to a compilation error

**Review by munterkalmsteiner — CHANGES_REQUESTED** [review]
There are two issues with this pull request:

1. It is not based on the master branch but on one of your own previous branches. You must have 1 branch for 1 thing, not mix different concerns together.
2. On the first sight, the code works. However, when I tested it, I also got some non-existing domains to work, for example http://example.com, see screenshot:
image

**Comment by katikireddy622** [comment]
@munterkalmsteiner  This is regarding the second issue in the pull request. [www.example.com](http://example.com) seems to be an existing domain. We have tried a non-existing domain called [www.katikireddy622.com](http://katikireddy622.com) in the first screenshot and since it does not exist it returns an error code 404. 

Screenshot (66)
Screenshot (67)

**Comment by munterkalmsteiner** [comment]
Interesting... I get different results here:
image

But ok, the code regarding the check looks ok.
Your code is still based on another branch, not on the master branch. This is not good practtice. If you have time, please fix it.

**Comment by munterkalmsteiner** [comment]
Overall, good effort!

**Pull request closed without merging**